src/main/java/org/altbeacon/beacon/BeaconManager.java

```diff
@@ -109,9 +109,9 @@
     protected static BeaconManager client = null;
     private final ConcurrentMap<BeaconConsumer, ConsumerInfo> consumers = new ConcurrentHashMap<BeaconConsumer,ConsumerInfo>();
     private Messenger serviceMessenger = null;
-    protected RangeNotifier rangeNotifier = null;
+    protected final List<RangeNotifier> rangeNotifiers = new CopyOnWriteArrayList<>();
```

Comment:
I think to use an [CopyOnWriteArraySet](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/CopyOnWriteArraySet.html). What do you think?

As consequences, the getMonitorNotifier() and the getRangeNotifier() method are irrelevants. Do you want to keep them or can I remove them ? 
(I think these methods are only useful internally to the library. They are not in usage any more inside the library)